cmd/bbolt/main.go

```diff
@@ -1796,46 +1799,62 @@ func ReadPage(path string, pageID int) (*page, []byte, error) {
 
 	// Determine total number of blocks.
 	p := (*page)(unsafe.Pointer(&buf[0]))
+	if p.id != pgid(pageID) {
+		return nil, nil, fmt.Errorf("error: %w due to unexpected page id: %d != %d", ErrCorrupt, p.id, pageID)
+	}
 	overflowN := p.overflow
+	if overflowN >= uint32(hwm) {
```

Comment:
We need to exclude the 2 meta pages and current page. For example, assuming there are 6 pages in total, then the overflowN field of any page should be <= 3

```suggestion
	if overflowN >= uint32(hwm) - 3 {
```

Reply:
Corrected. I didn't intended to be that precise - just protect OS from extensive allocation.